draft-ietf-lisp-pubsub.xml

```diff
@@ -342,15 +343,15 @@ Rec +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 
           <t>In the particular case of PubSub, cache poisoning via malicious Map-Notify messages is avoided by the use of nonce and the security association between the ITRs and the Map-Servers.</t>
 
-          <t>To prevent xTR-ID hijacking, it is RECOMMENDED to follow guidance from Section 9 of <xref target="RFC9301"></xref> to ensure integrity protection of Map-Request messages.</t>
+          <t>It is RECOMMENDED to follow guidance from the last paragraph of Section 9 of <xref target="RFC9301"></xref> to ensure integrity protection of Map-Request messages (e.g. to prevent xTR-ID hijacking).</t>
 
           <section anchor="association" title="Security Association between ITR and Map-Server">
 
             <t>Since Map-Notifies from the Map-Server to the ITR need to be authenticated, there is a need for a soft-state or hard-state security association (e.g., a PubSubKey) between the ITRs and the Map-Servers. For some controlled deployments, it might be possible to have a shared PubSubKey (or set of keys) between the ITRs and the Map-Servers. However, if pre-shared keys are not used in the deployment, LISP-SEC <xref target="RFC9303"></xref> can be used as follows to create a security association between the ITR and the MS.</t>
 
-            <t>First, when the ITR is sending a Map-Request with the N-bit set following <xref target="sub"></xref>, the ITR also performs the steps described in Section 5.4 of <xref target="RFC9303"></xref>. The ITR can then generate a PubSubKey by deriving a key from the One-Time Key (OTK) as follows: PubSubKey = KDF( OTK ), where KDF is the Key Derivation Function indicated by the OTK Wrapping ID. If OTK Wrapping ID equals NULL-KEY-WRAP-128 then the PubSubKey is the OTK. Note that as opposed to the pre-shared PubSubKey, this generated PubSubKey is different per EID-Prefix the ITR subscribes to (since the ITR will use a different OTK per Map-Request).</t>
+            <t>First, when the ITR is sending a Map-Request with the N-bit set following <xref target="sub"></xref>, the ITR also performs the steps described in Section 6.4 of <xref target="RFC9303"></xref>. The ITR can then generate a PubSubKey by deriving a key from the One-Time Key (OTK) and Map-Request nonce as follows: PubSubKey = KDF( OTK + nonce ), where KDF is the Key Derivation Function indicated by the OTK Wrapping ID. If OTK Wrapping ID equals NULL-KEY-WRAP-128 then the PubSubKey is the OTK. Note that as opposed to the pre-shared PubSubKey, this generated PubSubKey is different per EID-Prefix the ITR subscribes to (since the ITR will use a different OTK per Map-Request).</t>
```

Comment:
```suggestion
            <t>First, when the ITR is sending a Map-Request with the N-bit set following <xref target="sub"></xref>, the ITR also performs the steps described in Section 6.4 of <xref target="RFC9303"></xref>. The ITR can then generate a PubSubKey by deriving a key from the One-Time Key (OTK) and Map-Request's nonce as follows: PubSubKey = KDF(OTK + nonce), where KDF is the Key Derivation Function indicated by the OTK Wrapping ID. If OTK Wrapping ID equals NULL-KEY-WRAP-128, then the PubSubKey is the OTK. Note that as opposed to the pre-shared PubSubKey, this generated PubSubKey is different per EID-Prefix to which an ITR subscribes (since the ITR will use a different OTK per Map-Request).</t>
```